opensrp-chw-hps/src/main/java/org/smartregister/chw/hps/interactor/BaseHpsServiceVisitInteractor.java

```diff
@@ -90,7 +108,21 @@ private void evaluateHpsClientCriteria(Map<String, List<VisitDetail>> details) t
 
     private void evaluateHpsEducationOnBehavioralChange(Map<String, List<VisitDetail>> details) throws BaseHpsVisitAction.ValidationException {
 
-        HpsEducationOnBehaviouralChangeActionHelper actionHelper = new HpsEducationOnBehaviouralChangeActionHelper(mContext, memberObject);
+        HpsEducationOnBehaviouralChangeActionHelper actionHelper = new HpsEducationOnBehaviouralChangeActionHelper(mContext, memberObject){
+            @Override
+            public BaseHpsVisitAction.Status evaluateStatusOnPayload() {
+                if (StringUtils.isNotBlank(healthEducationProvided)) {
+                    try {
+                        new AppExecutors().mainThread().execute(() -> callBack.preloadActions(actionList));
+                        evaluateHpsRemarks(details);
+                    } catch(Exception e){
+                        Timber.e(e.toString());
+                    }
+                    return BaseHpsVisitAction.Status.COMPLETED;
+                }
+                return BaseHpsVisitAction.Status.PENDING;
+            }
+        };
```

Comment:
This part is not required. HpsClientCriteriaActionHelper is enough to achieve the intended action.
Ukiiweka hii part na pia kwenye other action helpers, contents za Comment Action zitakuwa overriden kila wakati mtu akijaza any other action